test/common/tracing/http_tracer_impl_test.cc

```diff
@@ -522,6 +522,9 @@ TEST_F(HttpConnManFinalizerImplTest, GrpcOkStatus) {
   EXPECT_CALL(span, setTag(Eq(Tracing::Tags::get().HttpMethod), Eq("POST")));
   EXPECT_CALL(span, setTag(Eq(Tracing::Tags::get().HttpProtocol), Eq("HTTP/2")));
   EXPECT_CALL(span, setTag(Eq(Tracing::Tags::get().HttpStatusCode), Eq("200")));
+  EXPECT_CALL(span, setTag(Eq(Tracing::Tags::get().GrpcPath), Eq("/pb.Foo/Bar")));
```

Comment:
Isn't this basically redundant to the `HttpUrl` tag? For perf reasons I don't think we need to duplicate this. Also, can you potentially remove the EXPECT_CALL at line 521 so it's easier to see exactly what spans we are getting in the test?

Reply:
It duplicates some of the data in `HttpUrl` (which would be `http://example.com:80/pb.Foo/Bar` in this case).
When observing individual traces, it could be considered redundant. But when analyzing collections of traces across multiple hosts/instances, it's easier when the tags can be directly used to group things. `HttpUrl` values aren't as easy to group as just the `:path` value.
That's not to dismiss the performance concerns, but to explain the way this data gets used.

Will push a change shortly to remove that catch-all `EXPECT_CALL`.

Reply:
I opened https://github.com/envoyproxy/envoy/issues/9653 to track the perf implications. We can merge this as is for now.